app/models/event.rb

```diff
@@ -38,8 +39,10 @@ def data_params
     @data_params ||= details[:data_params]
   end
 
+  # NB: Dynamic attributes for each event are all in the jsonb details column. Static attributes that are common and validated on all events are now flattened as individual columns in their own right. These include the client_timestamp and notes columns.
+  # This will be simplified once suppliers have migrated to the generic events structure.
   def notes
-    @notes ||= event_params.dig(:attributes, :notes)
+    @notes ||= super || event_params.dig(:attributes, :notes)
```

Comment:
`notes` is a column in the database. When you use STI it is common to resolve differences in data between implementations of  the parent event class by adding columns and have them affect all records. This becomes a burden/confusing to clients of the table because you end up having an instance of an Event (say MoveCancelV2) that ends up having lots of nil attributes when you pull it out.

In our implementation we're moving to using strictly common (shared by all types of Event) columns in a flat column, and dynamic/changing columns in a jsonb column.

So common to all Event implementations are:

1. client_timestamp
2. notes
3. type

And validated/different between events is:

1. details

For older versions of events we had the notes in the details jsonb.

```ruby
event.details.dig(:event_params, :attributes, :notes)
```
For new versions of events we have the notes in a flat column:

```ruby
event.notes
```

Reply:
Super just picks out the flat attribute (e.g. `event.notes`)

Reply:
it's clear now :)